DXMainClient/Domain/DiscordHandler.cs

```diff
@@ -68,23 +98,44 @@ public override void Initialize()
             client.OnSubscribe += OnSubscribe;
             client.OnUnsubscribe += OnUnsubscribe;
 
-            client.Initialize();
-            base.Initialize();
+            if (CurrentPresence != null)
+                client.SetPresence(CurrentPresence);
         }
 
-        public override void Update(GameTime gameTime)
+        /// <summary>
+        /// Connects to Discord.
+        /// Does not do anything if the Discord RPC client has not been initialized or is already connected.
+        /// </summary>
+        public void Connect()
         {
-            client.Invoke();
-            base.Update(gameTime);
+            if (client == null || client != null && client.IsInitialized)
+                return;
+
+            bool success = client.Initialize();
+
+            if (success)
+                Logger.Log("DiscordHandler: Connected Discord RPC client.");
+            else
+                Logger.Log("DiscordHandler: Failed to connect Discord RPC client.");
         }
 
-        protected override void Dispose(bool disposing)
+        /// <summary>
+        /// Disconnects from Discord.
+        /// Does not do anything if the Discord RPC client has not been initialized or is not connected.
+        /// </summary>
+        public void Disconnect()
         {
-            client.Dispose();
-            base.Dispose(disposing);
-        }
+            if (client == null || !client.IsInitialized)
+                return;
+
+            // HACK warning
+            // Currently DiscordRpcClient does not appear to have any way to reliably disconnect and reconnect using same client object.
+            // Deinitialize does not appear to completely reset connection state & resources and any attempts to call Initialize afterwards will fail.
+            // A hacky solution is to dispose current client object and create and initialize a new one.
+            InitializeClient(); //client.Deinitialize();
```

Comment:
Have you checked whether this is fixed in the new versions of the library?

Reply:
Yes, the issue is still present.